sky/skylet/log_lib.py

```diff
@@ -448,7 +460,12 @@ def tail_logs(job_id: Optional[int],
         try:
             start_stream = False
             with open(log_path, 'r', encoding='utf-8') as f:
-                for line in f.readlines():
+                if tail > 0:
+                    lines = deque(f.readlines(), maxlen=tail)
+                    start_stream = True
```

Comment:
Worth a comment for this. Also, if the number of lines is less than `tail`, it may cause additional lines to be printed with `start_stream` directly set to True?

Reply:
Yes, fixed

Reply:
Now peek the head of N lines to check if we need to delay `start_stream`